src/backend/agent.js

```diff
@@ -480,6 +531,31 @@ export default class Agent extends EventEmitter {
     //
     // this._bridge.send('operations', operations, [operations.buffer]);
     this._bridge.send('operations', operations);
+
+    if (this._persistedSelection !== null) {
```

Comment:
Sure, I can throttle them. However throttling implementation will likely involve the same number of checks. (One or two comparisons.) I’m not sure this optimization makes sense to me (unlike logic in traversals, this runs once per commit) but I can do it.

Reply:
Unless you mean you expect every commit to actually change the ID. That seems unlikely (usually many layers unlock at once) and even if it were to happen, it would only happen for at most N commits where N is the path length.

Reply:
> Sure, I can throttle them. However throttling implementation will likely involve the same number of checks.

I don't think I understand this statement.

Anecdotally,  I've seen a lot of apps that have really quick cascading updates in a small window of time. Seems like  we  could throttle the below calls to `getBestMatchForTrackedPath`  without any downside?

Reply:
>I don't think I understand this statement.

I'm saying that if your concern is the three or four equality checks that this block is doing, replacing them with a throttled call will likely also do a similar number of equality checks. Throttling involves getting the current time and comparing it to last call time. So I'm not sure what we're optimizing.

Seems like the concern is the `getBestMatchForTrackedPath` call and the while loop inside of it? Sure, let's throttle that.

Reply:
One additional difficulty here is that we can't naively "throttle" it because we don't want to miss any particular renderer's events. If two renderers commit in quick succession, we still want to ask both of them. I could throttle by renderer I guess but I honestly don't know that it would be faster than just doing the work eagerly.

Reply:
Ah, thanks for clarifying.

> Seems like the concern is the `getBestMatchForTrackedPath` call and the while loop inside of it? Sure, let's throttle that.

Yes, this was the concern.

Fair point about the throttling needing to be per-renderer. Meh. Ok.

Reply:
I've added instrumentation around this and simulated the worst case scenario: unlocking 99 levels but never managing to unblock the 100th one (because the innermost one never mounts) + doing a commit on every frame.

Over a 1000 calls of `onHookOperations` (~16 seconds of 60 commits per second each of which contains operations), the time we spend in this code block (in the worst case of always trying to restore a node) is a total of 4 milliseconds. That's 0.004 milliseconds per call.

If you still feel strongly I can add per-renderer throttling.

Reply:
I don't think it's necessary. I'm mostly just concerned about a "death from a thousand paper cuts" type of situation as we add more advanced features. I think this case is probably okay.

Reply:
>Yes, this was the concern.

One thing we could do is precaculate the ID eagerly at the time we match. Then the return value of `getBestMatchForTrackedPath` would be static. It's a bit more fragile but I don't mind doing that!